### DISCUSSION TITLE
how to install/use tailwindcss v2.1+

### DISCUSSION THREAD

**Discussion by playgroundctl** [tailwindlabs/tailwindcss#5183] [Help] [answered]
Hi,

and thank you! for making `tailwindcss` available and the application-ui with the very beautiful html components. 
It is what I was always needed (together with alpine.js) in the not-so-often occasions that I get to develop web apps.

I have a javascript-newbie question; 
I need to use the `filter grayscale` feature https://tailwindcss.com/docs/grayscale that requires v2.1+.

I have installed tailwindcss via `npm`, like this:
```
# npm init -y
# npm install -D tailwindcss postcss autoprefixer vite cssnano postcss-cli
# npm install @tailwindcss/forms
# npx tailwindcss init -p
# 1) package.json -> "dev": "vite"
# 2) tailwind.config.js -> require('@tailwindcss/forms'),  # https://github.com/tailwindlabs/tailwindcss-forms
# 3) mkdir {images,styles,scripts,templates}
# 4) cp node_modules/tailwindcss/tailwind.css styles/tailwind.css
# 5) touch index.html
# npm run dev
# N) postcss.config.js -> cssnano: {preset: 'default',},
```
and I build tailwindcss, like this:
```
# NODE_ENV=production npx tailwindcss-cli build styles/tailwind.css -o styles/glass.css
# NODE_ENV=production npx postcss styles/glass.css > styles/glass.min.css
```

All working fine so far. I am a happy tailwindcss user. But, when I try to install a higher (than 2.0.4) version 
```
% npm install -D tailwindcss@2.1

added 19 packages, removed 26 packages, changed 2 packages, and audited 231 packages in 3s

33 packages are looking for funding
  run `npm fund` for details

4 moderate severity vulnerabilities

To address all issues, run:
  npm audit fix --force

Run `npm audit` for details.
% npm install -D tailwindcss@2.2.7

added 26 packages, removed 19 packages, changed 2 packages, and audited 238 packages in 2s

36 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

When I try to build tailwindcss, `2.0.4` it is again ...
```
% NODE_ENV=production npx tailwindcss-cli build styles/tailwind.css -o styles/alpha.css
  
   tailwindcss 2.0.4
  
   🚀 Building: styles/tailwind.css
  
   ✅ Finished in 2.3 s
   📦 Size: 30.89KB
   💾 Saved to styles/alpha.css
```

Would you be so kind to explain what I am doing wrong? Why ` tailwindcss-cli` uses `2.0.4` instead of `2.2.7`?

* **package.json**
```
{
  "name": "beta",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "dev": "vite"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "description": "",
  "devDependencies": {
    "autoprefixer": "^10.3.1",
    "cssnano": "^5.0.7",
    "postcss": "^8.3.6",
    "postcss-cli": "^8.3.1",
    "tailwindcss": "^2.2.7",
    "vite": "^2.4.4"
  },
  "dependencies": {
    "@tailwindcss/forms": "^0.3.3"
  }
}
```

**Comment by Martleby** [comment]
I'd try: `npm install tailwindcss@latest`

That tends to work for me 👍

**Reply by playgroundctl** [reply]
thank you @Martleby for looking at it. from what I understood `@latest` installs `2.0.4` I tried that also.
what is puzzling me is that tailwindcss v2.1+ gets installed, see terminal output about the updates it receives when installing `2.2.7` (the latter appears correctly in `package.json` as well), **but** `tailwind-cli` uses `2.0.4` :(

**Reply by simonswiss** [reply]
Hey!

Gotta cook dinner for the kids so cannot reply to this right now, but as a very quick tip, Tailwind CSS v2.1 brings the CLI in core, so you shouldn't use `tailwind-cli` - you can run commands directly with `tailwindcss` like so:

```sh
tailwindcss build -o dist/tailwind.css --watch
```

I'll try look at the question in details later tonight or tomorrow and give a reply 👍

**Reply by playgroundctl** [reply]
oh thanks! that looks better
```
% ./node_modules/.bin/tailwindcss build -o styles/xxx.css --watch
Done in 6994ms.


```
the above command hasn't completed since ~30min now, the file `xxx.css` has been created though, what can be the reason, any ideas?

**Comment by playgroundctl** [comment, marked as answer]
re-reading the docs, I ended up doing :)
```
NODE_ENV=production npx postcss build styles/tailwind.css -o styles/glass.css
NODE_ENV=production npx postcss styles/glass.css > styles/glass.min.css
```
`grayscale` in `tailwind` works, awesome!
